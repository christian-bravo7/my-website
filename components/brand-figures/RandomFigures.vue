<template>
  <div
    class="absolute inset-0 | flex justify-center items-center | overflow-hidden pointer-events-none | random-figures"
  >
    <figure
      v-for="index in 10"
      :key="index"
      :class="`absolute | opacity-50 | primary-text figure figure--${index}`"
    >
      <component
        :is="figureComponent"
        class="w-full h-full | fill-current"
      />
    </figure>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';

@Component
export default class RandomFigures extends Vue {
  @Prop({ type: [String, Object as () => Vue], default: 'div' })
  readonly figureComponent!: string | Vue;
}
</script>

<style lang="scss" scoped>
.random-figures {
  z-index: -1;
}

.figure {
  &--1 {
    @apply w-12 h-12;

    top: 5%;
    left: 10%;
    transform: rotate(35deg);
  }

  &--2 {
    @apply w-12 h-12 hidden;

    top: 10%;
    left: 35%;
    transform: rotate(5deg);

    @screen md {
      @apply block;
    }
  }

  &--3 {
    @apply w-12 h-12;

    top: 90%;
    left: 50%;
    transform: rotate(0deg);
  }

  &--4 {
    @apply w-12 h-12;

    top: 80%;
    left: 0%;
    transform: rotate(45deg);
  }

  &--5 {
    @apply w-16 h-16;

    top: 20%;
    left: 80%;
    transform: rotate(25deg);
  }

  &--6 {
    @apply w-20 h-20 hidden;

    top: 65%;
    left: 75%;
    transform: rotate(-15deg);

    @screen md {
      @apply block;
    }
  }

  &--7 {
    @apply w-24 h-24;

    top: 65%;
    left: 90%;
    transform: rotate(20deg);
  }

  &--8 {
    @apply w-32 h-32;

    top: 95%;
    left: 20%;
    transform: rotate(15deg);
  }

  &--9 {
    @apply w-40 h-40 hidden;

    top: 5%;
    left: 90%;
    transform: rotate(35deg);

    @screen md {
      @apply block;
    }
  }

  &--10 {
    @apply w-48 h-48;

    top: 35%;
    left: 4%;
    transform: rotate(-25deg);
  }
}
</style>
