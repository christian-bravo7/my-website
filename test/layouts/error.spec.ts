import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';

import Error from '@/layouts/error.vue';
import RandomFigures from '@/components/app-figures/RandomFigures.vue';

import setupI18n from '@/test/config/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('error page', () => {
  it('should render correclty', () => {
    const wrapper = mount(Error, {
      stubs: {
        NuxtLink: RouterLinkStub,
        Nuxt: {
          template: '<div></div>',
        },
      },
      components: {
        RandomFigures,
      },
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
