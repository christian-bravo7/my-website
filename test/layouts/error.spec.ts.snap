// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error page should render correclty 1`] = `
<section class="flex items-center justify-center | w-full">
  <section class="flex flex-col items-center justify-center"><span id="oops-label" class="text-5xl md:text-8xl | mb-6 | text-gray-900 dark:text-gray-100 | font-bold">
      @#$%@#
    </span>
    <p class="text-md md:text-3xl | text-center | mb-10 md:mb-12 | text-gray-900 dark:text-gray-100">
      Sorry, but this page doesn't exist.
    </p> <a class="px-6 py-3 | text-base md:text-lg | rounded-full | bg-pink-500 dark:bg-blue-500 | text-white">
      Take me home
    </a>
  </section>
  <div class="absolute inset-0 | flex justify-center items-center | overflow-hidden pointer-events-none | random-figures">
    <figure class="absolute | opacity-25 | primary-text figure figure--1"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--2"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--3"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--4"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--5"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--6"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--7"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--8"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--9"><svg class="w-full h-full | fill-current"></svg></figure>
    <figure class="absolute | opacity-25 | primary-text figure figure--10"><svg class="w-full h-full | fill-current"></svg></figure>
  </div>
</section>
`;
